test/OpenTelemetry.Tests/Metrics/MetricExemplarTests.cs

```diff
@@ -0,0 +1,212 @@
+            var metricPoint = GetFirstMetricPoint(exportedItems);
+            Assert.NotNull(metricPoint);
+            Assert.True(metricPoint.Value.StartTime >= testStartTime);
+            Assert.True(metricPoint.Value.EndTime != default);
+            var exemplars = GetExemplars(metricPoint.Value);
+            ValidateExemplars(exemplars, metricPoint.Value.StartTime, metricPoint.Value.EndTime, measurementValues, false);
+
+            exportedItems.Clear();
+
+#if NETFRAMEWORK
+            Thread.Sleep(10); // Compensates for low resolution timing in netfx.
+#endif
+
+            measurementValues = GenerateRandomValues(10);
+            foreach (var value in measurementValues)
+            {
+                var act = new Activity("test").Start();
+                histogram.Record(value);
+                act.Stop();
+            }
+
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+            metricPoint = GetFirstMetricPoint(exportedItems);
+            Assert.NotNull(metricPoint);
+            Assert.True(metricPoint.Value.StartTime >= testStartTime);
+            Assert.True(metricPoint.Value.EndTime != default);
+            exemplars = GetExemplars(metricPoint.Value);
+            ValidateExemplars(exemplars, metricPoint.Value.StartTime, metricPoint.Value.EndTime, measurementValues, true);
+        }
+
+        [Fact]
+        public void TestExemplarsFilterTags()
+        {
+            DateTime testStartTime = DateTime.UtcNow;
+            var exportedItems = new List<Metric>();
+
+            using var meter = new Meter($"{Utils.GetCurrentMethodName()}");
+            var histogram = meter.CreateHistogram<double>("testHistogram");
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .SetExemplarFilter(new AlwaysOnExemplarFilter())
+                .AddView(histogram.Name, new MetricStreamConfiguration() { TagKeys = new string[] { "key1" } })
+                .AddInMemoryExporter(exportedItems, metricReaderOptions =>
+                {
+                    metricReaderOptions.TemporalityPreference = MetricReaderTemporalityPreference.Delta;
+                })
+                .Build();
+
+            var measurementValues = GenerateRandomValues(10);
+            foreach (var value in measurementValues)
+            {
+                histogram.Record(value, new("key1", "value1"), new("key2", "value1"), new("key3", "value1"));
+            }
+
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+            var metricPoint = GetFirstMetricPoint(exportedItems);
+            Assert.NotNull(metricPoint);
+            Assert.True(metricPoint.Value.StartTime >= testStartTime);
+            Assert.True(metricPoint.Value.EndTime != default);
+            var exemplars = GetExemplars(metricPoint.Value);
+            Assert.NotNull(exemplars);
+            var filteredTags = exemplars[0].FilteredTags;
```

Comment:
We should probably test for correctness of `AlignedHistogramBucketExemplarReservoir` by checking the exact exmplar index that we expect to be populated. That'll be good to avoid any regression in the future but testing that doesn't seem as straightforward.

Reply:
yea. will be doing it as part of targetted tests for the built-in exemplarreservoirs.